torch/csrc/jit/codegen/cuda/kernel.cpp

```diff
@@ -155,6 +155,139 @@ std::pair<std::string, std::string> codeGeneration(Fusion& fusion) {
   return std::make_pair(func_name, str_stream.str());
 };
 
+bool validateKernelArgTensor(
+    const at::Tensor& arg,
+    const Val& param,
```

Comment:
It's typical to pass in parameters as const references for parameters that are not expected to change, however, in this code base we have tended to pass Val Objects as pointers.  This is mostly a style question if the deviation matters here?

Reply:
I prefer const references when possible as they are a little safer.

Reply:
I still don't follow how they are safer?
`
void myFunc(const Val& val){
  val->member_func();
}
Val* null_val = nullptr;
myFunc(*null_val);`
Will still produce a segfault.

If you're worried about constness, then
`void myFunc(const Val* const val){...}`
will give you all the same const protection.

However, you shouldn't worry about the constness of Vals generally, as their public mutability is carefully designed. All Vals are immutable except for interface nodes. Primarily TensorViews and their Domains.

I'd like a stronger reason to start using const refs, otherwise the convention is pointers.